### PR TITLE
R: Fix warning on mismatched integer comparison

### PR DESCRIPTION
**Pull request by carlopi**
Solves CI failure on [R Package Windows (Warnings)](https://github.com/duckdb/duckdb/actions/runs/5908316450/job/16027906274#logs)

### CODE REVIEW

File: tools/rpkg/src/types.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ Vector RType::GetFactorLevels() const {
 	D_ASSERT(id_ == RTypeId::FACTOR);
 	Vector duckdb_levels(LogicalType::VARCHAR, aux_.size());
 	auto levels_ptr = FlatVector::GetData<string_t>(duckdb_levels);
-	for (R_xlen_t level_idx = 0; level_idx < aux_.size(); level_idx++) {
+	for (R_xlen_t level_idx = 0; (size_t)level_idx < aux_.size(); level_idx++) {
```
Comment (krlmlr) [inline tools/rpkg/src/types.cpp:43]: No reason to use `R_xlen_t` at all:

```suggestion
	for (size_t level_idx = 0; level_idx < aux_.size(); ++level_idx) {
```

**Review by krlmlr — APPROVED** [review]
Thanks for catching it, missed that.

**Comment by carlopi** [comment]
Perfect, this is ready to go then!

File: tools/rpkg/src/types.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ Vector RType::GetFactorLevels() const {
 	D_ASSERT(id_ == RTypeId::FACTOR);
 	Vector duckdb_levels(LogicalType::VARCHAR, aux_.size());
 	auto levels_ptr = FlatVector::GetData<string_t>(duckdb_levels);
-	for (R_xlen_t level_idx = 0; level_idx < aux_.size(); level_idx++) {
+	for (size_t level_idx = 0; level_idx < aux_.size(); ++level_idx) {
```
Comment (hannes) [inline tools/rpkg/src/types.cpp:43]: why switch from `level_idx++` to `++level_idx`?

**Comment by carlopi** [comment]
I ended up cleaning the PR, to simply changing from `R_xlen_t` to `size_t`.

**Pull request merged**